lib/libvarnishapi/vsl_cursor.c

```diff
@@ -335,18 +335,20 @@ vslc_file_delete(const struct VSL_cursor *cursor)
 
 /* Read n bytes from fd into buf */
 static ssize_t
-vslc_file_readn(int fd, void *buf, ssize_t n)
+vslc_file_readn(int fd, void *bufa, ssize_t n)
 {
-	ssize_t t = 0;
+	char *buf = bufa;
 	ssize_t l;
 
-	while (t < n) {
-		l = read(fd, (char *)buf + t, n - t);
+	while (n > 0) {
+		l = read(fd, buf, n);
 		if (l <= 0)
 			return (l);
-		t += l;
+		assert(l <= n);
```

Comment:
This one was not introduced here, but shouldn't we treat this as a short read if `t` is greater than zero?

Reply:
IIRC we can have short reads with some file systems

Reply:
You mean `return (pdiff(...))` also for the `l <= 0` case? I thought about it, but did not want to change this behavior in case it mattered.